src/store/global.ts

```diff
@@ -0,0 +1,36 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { FilterOption } from '../components/MailList';
+import { accountLogout } from './account';
+
+interface GlobalState {
+  messageListFilters: MessageListFilters;
+}
+
+const initialState: GlobalState = {
+  messageListFilters: {},
+};
+
+interface MessageListFilters {
+  [filterType: string]: string | FilterOption;
+}
+
+export const globalSlice = createSlice({
+  name: 'global',
+  initialState,
+  reducers: {
+    setMessageListFilters(state, action: PayloadAction<MessageListFilters>) {
```

Comment:
could you please rename action like "update" or something, it's ease to confuse with setState function